.github/workflows/actions.yml

```diff
@@ -0,0 +1,38 @@
+# This workflow will install Python dependencies and run some pytests
+name: Python application
+
+on:
+  push:
+    branches: [ "develop" ]
+  pull_request:
+    branches: [ "develop" ]
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - uses: actions/cache@v3
+      id: cache
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.*') }}
+        restore-keys: | 
+          ${{ runner.os }}-pip-
```

Comment:
what is the cache for? genuinely curious :)

Reply:
It will cache the pip packages so they must not be downloaded every time. 
See here for details: https://earthly.dev/blog/caching-dependencies-github-actions/

Reply:
wooow! That is cool haha